eval/src/vespa/eval/eval/dump_target.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <vespa/vespalib/stllike/string.h>
+
+namespace vespalib {
+namespace eval {
+
+struct TensorFunction;
+struct DumpTargetBackend;
+enum class Aggr;
+
+class DumpTarget
+{
+private:
+    DumpTargetBackend &_back_end;
+    DumpTarget(DumpTargetBackend &back_end);
+    DumpTarget(DumpTargetBackend &back_end, size_t level);
+    size_t _indentLevel;
+    vespalib::string _nodeName;
+    void indent();
+public:
+    static vespalib::string dump(const TensorFunction &root);
```

Comment:
why not just expose this function and make the rest implementation details? and call the files 'dump_tensor_function.{h,cpp}' to match similar operations.